crates/turborepo-lib/src/run/summary/task.rs

```diff
@@ -74,3 +94,94 @@ impl<'a> TaskSummary<'a> {
         self.package = None;
     }
 }
+
+impl TaskCacheSummary {
+    pub fn cache_miss() -> Self {
+        Self {
+            local: false,
+            remote: false,
+            status: CacheStatus::Miss,
+            time_saved: 0,
+            source: None,
+        }
+    }
+}
+
+impl From<Option<CacheResponse>> for TaskCacheSummary {
+    fn from(value: Option<CacheResponse>) -> Self {
+        value.map_or_else(Self::cache_miss, |CacheResponse { source, time_saved }| {
+            let source = CacheSource::from(source);
+            // Assign these deprecated fields Local and Remote based on the information
+            // available in the itemStatus. Note that these fields are
+            // problematic, because an ItemStatus isn't always the composite
+            // of both local and remote caches. That means that an ItemStatus might say it
+            // was a local cache hit, and we return remote: false here. That's misleading
+            // because it does not mean that there is no remote cache hit,
+            // it _could_ mean that we never checked the remote cache. These
+            // fields are being deprecated for this reason.
+            let (local, remote) = match source {
```

Comment:
Yeah this setup is problematic for a few reasons. For one, I regret making a cache miss an error instead of just another variant. That's my bad. Second, booleans are kinda misrepresentative because `local: true` and `remote: false` doesn't mean we failed to get a remote cache hit, it just means we didn't even try because the local cache had it (why would you fetch from remote when you can fetch from local?). For 2.0, we should consider changing this schema

Reply:
Most definitely, this logic and comment were copied from Go